contracts/BatchExchange.sol

```diff
@@ -291,9 +292,9 @@ contract BatchExchange is EpochTokenLocker {
         require(utility.add(burntFees) > disregardedUtility, "Solution must be better than trivial");
         // burntFees ensures direct trades (when available) yield better solutions than longer rings
         uint256 objectiveValue = utility.add(burntFees).sub(disregardedUtility);
+        tokenConservation.checkTokenConservation();
```

Comment:
conceptually I feel this change should be all the way up just after the tokenConservation has been calculated. I understand that this is not possible as some of our test cases wouldn't assert in the right spot, but I think this is a smell with our test suite.

Reply:
I reverted back to where it originally was.

Reply:
I moved this back to where it was.